platform/lib/pipeline/componentReferenceDocument.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const MarkdownDocument = require('./markdownDocument.js');
+
+const DEFAULT_VERSION = 0.1;
+const EXTENSION_TYPE_ELEMENT = 'element';
+const EXTENSION_TYPE_TEMPLATE = 'template';
+
+
+class ComponentReferenceDocument extends MarkdownDocument {
+  constructor(path, contents, extension) {
+    super(path, contents);
+
+    if (!this.teaser.text) {
+      throw new Error('Teaser missing in', extension.name);
```

Comment:
Was it discussed that documents without a teaser text should be skipped completely?

Reply:
As before a teaser text is now automatically extracted.